### PR TITLE
[GsoC][BanyanDB] Self-Observability: Run Time Modes 

### PR DESCRIPTION
**Pull request by Sylvie-Wxr**
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.
- [ ] Update the documentation to include this new feature.
- [ ] Tests(including UT, IT, E2E) are added to verify the new feature.
- [ ] If it's UI related, attach the screenshots below.
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Fixes apache/skywalking#10634.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking-banyandb/blob/main/CHANGES.md).

Implementation details:
1. Add observability modes flag 
2. Register providers to MetricsCollector at run time 
3. Select MetricsServerInterceptor at run time
4. remove all BUILD_TAGS=prometheus

TO-DO:
update documentation

### CODE REVIEW

File: banyand/observability/service.go [hunk 1]
```diff
@@ -52,19 +71,41 @@ type metricService struct {
...
 	flagSet.StringVar(&p.listenAddr, "observability-listener-addr", ":2121", "listen addr for observability")
+	flagSet.StringArrayVar(&p.modes, "observability-modes", []string{"prometheus"}, "modes for observability")
 	return flagSet
 }
 
 func (p *metricService) Validate() error {
 	if p.listenAddr == "" {
 		return errNoAddr
 	}
+	set := make(map[string]bool)
```
Comment (hanahmily) [inline banyand/observability/service.go:89]:
```suggestion
	set := make(map[string]struct{})
```

**Review by hanahmily — CHANGES_REQUESTED** [review]
Remove "BUILD_TAGS=prometheus" in "obervability.md" as well.

**Comment by hanahmily** [comment]
I also removed the default internal group, "self-observability." Please create this group when "nativeProvider" is active.

File: banyand/observability/service.go (hunk 1 above)
Comment (Sylvie-Wxr) [reply banyand/observability/service.go]: updated.

**Comment by Sylvie-Wxr** [comment]
> Remove "BUILD_TAGS=prometheus" in "obervability.md" as well.

Done. Updated documentation.

**Comment by Sylvie-Wxr** [comment]
> Remove "BUILD_TAGS=prometheus" in "obervability.md" as well.

Done. Updated documentation.

**Review by hanahmily — APPROVED** [review]

**Pull request merged**